src/frontend/src/utils/utils.ts

```diff
@@ -229,14 +229,48 @@ export class Chan<A> implements AsyncIterable<A> {
     }
   }
 
+  // Signal to `map` that the element should remain unchanged
+  static readonly unchanged = Symbol("unchanged");
+
   // Return a new Chan mapped with `f`.
-  map<B>(f: (a: A) => B): Chan<B> {
-    const latest = this.latest === undefined ? undefined : f(this.latest);
+  // In the simplest case, a mapping function is provided.
+  // For advanced cases, the mapping function may return 'Chan.unchanged' signalling
+  // that the element shouldn't be changed, in which case a default (initial) value
+  // also needs to be provided.
+  map<B>(
+    opts: ((a: A) => B) | { f: (a: A) => B | typeof Chan.unchanged; def: B }
+  ): Chan<B> {
+    // How the mapped chan should handle the value
+    let handleValue: (opts: { send: (b: B) => void; value: A }) => void;
+    let latest: B;
+
+    if (typeof opts === "function") {
```

Comment:
nth: would be nice to have this logic, then if / else / assign value in a function. 

Like:

```
const {handledValue, latest} = initHandledValue(...);
```

Reply:
ah yes, that'd be much nicer!